src/EditorFeatures/Test2/IntelliSense/VisualBasicCompletionCommandHandlerTests.vb

```diff
@@ -1249,22 +1289,28 @@ End Module|}          </Document>)
                 Dim subjectBuffer = subjectDocument.GetTextBuffer()
 
                 state.SendTypeCharsToSpecificViewAndBuffer("(", view, subjectBuffer)
-                Await state.AssertCompletionSession()
+                Await state.AssertCompletionSession(view)
                 state.SendTypeCharsToSpecificViewAndBuffer("a", view, subjectBuffer)
                 Await state.WaitForAsynchronousOperationsAsync()
-                Await state.AssertSelectedCompletionItem(displayText:="arg")
+                Await state.AssertSelectedCompletionItem(displayText:="arg", projectionsView:=view)
 
                 Dim text = view.TextSnapshot.GetText()
                 Dim projection = DirectCast(topProjectionBuffer.TextBuffer, IProjectionBuffer)
                 Dim sourceSpans = projection.CurrentSnapshot.GetSourceSpans()
 
                 ' unmap our source spans without changing the top buffer
-                projection.ReplaceSpans(0, sourceSpans.Count, {text}, EditOptions.DefaultMinimalChange, editTag:=Nothing)
+                projection.ReplaceSpans(0, sourceSpans.Count, {subjectBuffer.CurrentSnapshot.CreateTrackingSpan(0, subjectBuffer.CurrentSnapshot.Length, SpanTrackingMode.EdgeInclusive)}, EditOptions.DefaultMinimalChange, editTag:=Nothing)
+
+                If completionImplementation = CompletionImplementation.Legacy Then
+                    ' Make sure completionImplementation updates even though the subject buffer is not connected.
+                    Dim editorOperations = state.GetService(Of IEditorOperationsFactoryService).GetEditorOperations(view)
+                    editorOperations.Backspace()
+                    editorOperations.InsertText("b")
```

Comment:
The test is still skipped in the new completion because Editor does not handle EditorOperations when processing completion. Therefore, we cannot use EditorOperation in the new test.

To make a similar test, we need to find to what view/buffer should we send backspace and chars. There is an open issue tracking this: [https://github.com/dotnet/roslyn/issues/27446](https://github.com/dotnet/roslyn/issues/27446)

---
In reply to: [233123597](https://github.com/dotnet/roslyn/pull/29016#discussion_r233123597) [](ancestors = 233123597)